devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusRemoveExtension.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.gradle.tasks;
+
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.List;
+import java.util.Set;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+
+import io.quarkus.cli.commands.RemoveExtensions;
+import io.quarkus.cli.commands.file.GradleBuildFile;
+import io.quarkus.cli.commands.writer.FileProjectWriter;
+
+public class QuarkusRemoveExtension extends QuarkusPlatformTask {
+
+    public QuarkusRemoveExtension() {
+        super("Removes Quarkus extensions specified by the user to the project.");
+    }
+
+    private List<String> extensionsToRemove;
+
+    @Option(option = "extensions", description = "Configures the extensions to be removed.")
+    public void setExtensionsToRemove(List<String> extensionsToRemove) {
+        this.extensionsToRemove = extensionsToRemove;
+    }
+
+    @Input
+    public List<String> getExtensionsToRemove() {
+        return extensionsToRemove;
+    }
+
+    @TaskAction
+    public void removeExtension() {
+        Set<String> extensionsSet = getExtensionsToRemove()
+                .stream()
+                .flatMap(ext -> stream(ext.split(",")))
+                .map(String::trim)
+                .collect(toSet());
+        try {
+            GradleBuildFile gradleBuildFile = getProject().getParent() == null
+                    ? new GradleBuildFile(new FileProjectWriter(getProject().getProjectDir()))
+                    : new GradleBuildFile(new FileProjectWriter(getProject().getProjectDir()),
+                            new FileProjectWriter(getProject().getRootProject().getProjectDir()));
```

Comment:
We should really move this code to a single place

Reply:
Moved into a method in QuarkusPlatformTask